sdk/core/Azure.Core/tests/HttpPipelineFunctionalTests.cs

```diff
@@ -238,8 +238,9 @@ async Task Connect()
             await Task.WhenAll(requests);
         }
 
-        [Test]
-        public async Task BufferedResponsesReadableAfterMessageDisposed()
+        [TestCase(200)]
+        [TestCase(404)]
```

Comment:
what's significant about these two status codes?

Reply:
On .NET Framework the HttpWebRequest transport always pre-buffers the error responses. This causes our buffering policy to opt-out. There was a bug where we accidentally disposed the buffered body in this case that was fixed in https://github.com/Azure/azure-sdk-for-net/pull/23249.

This PR adds a test and cleans up the implementation a bit.

Reply:
Makes sense - I was just curious why we chose to test these specific response codes and why any response code is relevant to the test.